swarm/src/lib.rs

```diff
@@ -1580,8 +1580,8 @@ impl fmt::Display for DialError {
                 f,
                 "Dial error: tried to dial local peer id at {endpoint:?}."
             ),
-            DialError::DialPeerConditionFalse(c) => {
-                write!(f, "Dial error: condition {c:?} for dialing peer was false.")
+            DialError::DialPeerConditionFalse(_) => {
+                write!(f, "Dial error: condition dial was configured to only happen when disconnected (`PeerCondition::Disconnected`), but node is already connected, thus cancelling new dial.")
```

Comment:
I think there is more needed, i.e. one needs to differentiate on the `DialPeerConditionFalse` error. `PeerCondition::Disconnected` is one of the reasons it might fail. Take a look at the function throwing the `DialPeerConditionFalse`.